app/http.php

```diff
@@ -0,0 +1,353 @@
+    }
+
+    return $balancing;
+}, ['state', 'algo']);
+
+function fetchExecutorsState(Registry $register, bool $forceShowError = false): void
+{
+    $state = $register->get('state');
+
+    $executors = \explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
+
+    $health = new Health();
+
+    foreach ($executors as $executor) {
+        $health->addNode(new Node($executor));
+    }
+
+    $nodes = $health
+        ->run()
+        ->getNodes();
+
+    foreach ($nodes as $node) {
+        $status = $node->isOnline() ? 'online' : 'offline';
+        $hostname = $node->getHostname();
+
+        $oldStatus = $state->exists($hostname) ? $state->get($hostname)['status'] ?? null : null;
+        if ($forceShowError === true || ($oldStatus !== null && $oldStatus !== $status)) {
+            Console::success('Executor "' . $node->getHostname() . '" went ' . $status . '.');
+        }
+
+        $state->set($node->getHostname(), [
+            'status' => $status,
+            'hostname' => $hostname,
+            'state' => \json_encode($node->getState())
+        ]);
+    }
+}
+
+function logError(Throwable $error, string $action, ?Logger $logger, Utopia\Route $route = null): void
+{
+    if ($logger) {
+        $version = (string) App::getEnv('OPEN_RUNTIMES_PROXY_VERSION', 'UNKNOWN');
+
+        $log = new Log();
+        $log->setNamespace('proxy');
+        $log->setServer(\gethostname() ? \gethostname() : 'unknown');
+        $log->setVersion($version);
+        $log->setType(Log::TYPE_ERROR);
+        $log->setMessage($error->getMessage());
+
+        if ($route) {
+            $log->addTag('method', $route->getMethod());
+            $log->addTag('url', $route->getPath());
+        }
+
+        $log->addTag('code', $error->getCode());
+        $log->addTag('verboseType', get_class($error));
+        $log->addExtra('file', $error->getFile());
+        $log->addExtra('line', $error->getLine());
+        $log->addExtra('trace', $error->getTraceAsString());
+        $log->addExtra('detailedTrace', $error->getTrace());
+        $log->setAction($action);
+        $log->setEnvironment(App::isProduction() ? Log::ENVIRONMENT_PRODUCTION : Log::ENVIRONMENT_STAGING);
+
+        $responseCode = $logger->addLog($log);
+        Console::info('Proxy log pushed with status code: ' . $responseCode);
+    }
+
+    Console::error('[Error] Type: ' . get_class($error));
+    Console::error('[Error] Message: ' . $error->getMessage());
+    Console::error('[Error] File: ' . $error->getFile());
+    Console::error('[Error] Line: ' . $error->getLine());
+}
+
+\go(function () use ($register) {
+    if (App::getEnv('OPEN_RUNTIMES_PROXY_HEALTHCHECK', 'enabled') === 'enabled') {
+        fetchExecutorsState($register, true);
+    } else {
+        $state = $register->get('state');
+
+        // If no health check, mark all as online
+        $executors = \explode(',', (string) App::getEnv('OPEN_RUNTIMES_PROXY_EXECUTORS', ''));
```

Comment:
Each health check is updating the `state` anyway. We can have this entire block as part of `fetchExecutorsState`

Reply:
I wanted to do this case outside of `fetchExecutorsState`, since this is not really fetching the states. It's just marking everything offline because fetching is disabled.

Reply:
You mean online?

Reply:
*Online yes